---
 grub-core/term/efi/console.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/grub-core/term/efi/console.c
+++ b/grub-core/term/efi/console.c
@@ -124,9 +124,12 @@ grub_console_getkey (struct grub_term_in
   if (status != GRUB_EFI_SUCCESS)
     return GRUB_TERM_NO_KEY;
 
-  if (key.scan_code == 0)
-    return key.unicode_char;
-  else if (key.scan_code < ARRAY_SIZE (efi_codes))
+  if (key.scan_code == 0) {
+    if (key.unicode_char < 0x20 && key.unicode_char != 0 && key.unicode_char != '\t' && key.unicode_char != '\b' && key.unicode_char != '\n' && key.unicode_char != '\r')
+        return GRUB_TERM_CTRL | (key.unicode_char - 1 + 'a');
+    else
+        return key.unicode_char;
+  } else if (key.scan_code < ARRAY_SIZE (efi_codes))
     return efi_codes[key.scan_code];
 
   return GRUB_TERM_NO_KEY;
