From 542f41a8bdc385ed849170565ac353956a47683a Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Fri, 9 Jun 2017 12:45:21 +0300
Subject: [PATCH] Use rpm memory allocator so we dont need to check for return

---
 rpmsign.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/rpmsign.c b/rpmsign.c
index 35c5ee966..a59f2dc1c 100644
--- a/rpmsign.c
+++ b/rpmsign.c
@@ -59,14 +59,9 @@ static struct poptOption optionsTable[] = {
 static char *get_fskpass(void)
 {
     struct termios flags, tmp_flags;
-    char *password, *pwd;
     int passlen = 64;
-
-    password = malloc(passlen);
-    if (!password) {
-	perror("malloc");
-	return NULL;
-    }
+    char *password = xmalloc(passlen);
+    char *pwd;
 
     tcgetattr(fileno(stdin), &flags);
     tmp_flags = flags;
-- 
2.11.0

