From 104146d812a5591738235699b02fc8ae3fc44743 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui at cn.fujitsu.com>
Date: Mon, 17 Aug 2015 13:53:28 +0900
Subject: [PATCH] fix the cflags for all of targets.

Signed-off-by: Lei Maohui <leimaohui at cn.fujitsu.com>
---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index 824bbbf..d24cc44 100644
--- a/Makefile
+++ b/Makefile
@@ -64,7 +64,6 @@ USRLIBDIR	:= $(patsubst /lib/%,/usr/lib/%,$(LIBDIR))
 endif
 BUILDFOR	:= $(shell file /usr/bin/make | sed -e 's!.*ELF \(32\|64\)-bit.*!\1!')-bit
 
-ifeq ($(origin CFLAGS),undefined)
 ifeq ($(BUILDFOR),32-bit)
 CFLAGS		+= -m32
 LIBDIR		:= /lib
@@ -76,7 +75,6 @@ LIBDIR		:= /lib64
 USRLIBDIR	:= /usr/lib64
 endif
 endif
-endif
 
 ###############################################################################
 #
-- 
2.7.4

